dfdaemon/downloader/downloader.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 1999-2017 Alibaba Group.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downloader
+
+// Interface specifies on how an plugin can download a file.
+type Interface interface {
```

Comment:
I do not think we should name an interface by `Interface`. It is not a good way.

We can use a more specific and readable to name an interface. For this part, I think `type Downloader interface` is OK.

For those structs who implements this interface, we could name that by `XXXDownloader` or `XXX` or something like that.

Reply:
If you read this as `downloader.Interface`(plus the package name), it is actually easy to understand, and such pattern is widely used in go projects(k/k as an example), too.

If we say it as `downloader.Downloader` which is pretty awkward, right?

Reply:
see https://golang.org/pkg/sort/#Interface